pkg/registry/core/pod/storage/eviction.go

```diff
@@ -325,12 +326,22 @@ func addConditionAndDeletePod(r *EvictionREST, ctx context.Context, name string,
 		}
 
 		podCopyUpdated := rest.DefaultUpdatedObjectInfo(pod, conditionAppender)
+		podCopyUpdatedWithDeletePreconditions := &updatedObjectInfoWithOverridePreconditions{podCopyUpdated, deleteOptions.Preconditions}
```

Comment:
I don't understand why we need a custom updatedObjectInfoWithOverridePreconditions wrapper / impl here for the update ... if deleteOptions.Preconditions.ResourceVersion and pod.ResourceVersion are both set and are different, we can reject with a conflict error without proceeding to update or delete, right?

Reply:
Thanks for the suggestion. I was just looking for a way to pass the preconditions somehow, but setting the values directly is more pragmatic and works as well. And I put a comment in the code to make it clearer.